odoo/addons/base/models/res_partner.py

```diff
@@ -261,6 +263,12 @@ def _compute_partner_share(self):
         for partner in self:
             partner.partner_share = not partner.user_ids or not any(not user.share for user in partner.user_ids)
 
+    @api.depends('vat')
+    def _compute_same_vat(self):
+        for partner in self:
+            partner.same_vat = partner.vat and self.env['res.partner'].search([('vat', '=', partner.vat)], limit=1) or False
+            partner.display_warning_vat = bool(partner.same_vat) and bool(partner.vat) and partner.same_vat != partner
```

Comment:
given above modif, we can now simplify
`partner.display_warning_vat = bool(partner.same_vat)`

and then we realize that field isn't needed since we can base the attrs on partner.same_vat directly

Reply:
Nope, we need to check that `partner.vat` is set too, otherwise it will always be displayed when no vat is set.
I can set the attributes as follow:
`invisible="bool(same_vat) and bool(vat)"`
or `attrs="{'invisible: [('display_partner_vat', '=', False), ('vat', '!=', False), ('vat', '!=', '')]}"`
But I don't think those syntax’s are better

Reply:
you can check partner.vat in the onchange, and fill same_vat_partner_id only if vat is given

Reply:
Ok, I also had to check the parents because partners with parent_id set have the TIN of their parent.
And I had to put `partner.id or -1` because `partner.id` can give ans instance of `NewId` in the form.